benchmarks/build.gradle

```diff
@@ -58,55 +66,65 @@ task jmhProfilers(type: JavaExec, description:'Lists the available profilers for
-  }
+  @TaskAction
+  void exec() {
+    main = 'org.openjdk.jmh.Main'
 
-  def exclude = project.properties.get('exclude');
-  def format = project.properties.getOrDefault('format', 'json');
-  def profilers = project.properties.get('profilers');
-  def jvmArgs = project.properties.get('jvmArgs')
-  def verify =  project.properties.get('verify');
+    def include = ''
+    if (project.hasProperty('include')) {
+      include = [".*" + project.properties.get('include') + ".*"]
+    }
+    if (project.hasProperty('fullInclude')) {
+      include = project.properties.get('fullInclude');
+    }
 
-  def resultFile = file("build/reports/jmh/result.${format}")
+    def exclude = project.properties.get('exclude');
+    def format = project.properties.getOrDefault('format', 'json');
+    def profilers = project.properties.get('profilers');
+    def jvmArgs = project.properties.get('jvmArgs')
+    def verify =  project.properties.get('verify');
 
-  args include
-  if(exclude) {
-    args '-e', exclude
-  }
-  if(verify != null) { // execute benchmarks with the minimum amount of execution (only to check if they are working)
-    println "≥≥ Running in verify mode"
-    args '-f' , 1
-    args '-wi' , 1
-    args '-i' , 1
-  }
-  args '-foe', 'true'   //fail-on-error
-  args '-v', 'NORMAL'   //verbosity [SILENT, NORMAL, EXTRA]
-  if(profilers) {
-    profilers.split(',').each {
-      args '-prof', it
+    def resultFile = project.file("build/reports/${this.name}/result.${format}")
+
+    args include
+    if(exclude) {
+      args '-e', exclude
     }
-  }
-  args '-jvmArgsPrepend', '-Xmx3072m'
-  args '-jvmArgsPrepend', '-Xms3072m'
-  if(jvmArgs) {
-    for(jvmArg in jvmArgs.split(' ')) {
-      args '-jvmArgsPrepend', jvmArg
+    if(verify != null) { // execute benchmarks with the minimum amount of execution (only to check if they are working)
+      println "≥≥ Running in verify mode"
+      args '-f' , 1
+      args '-wi' , 1
+      args '-i' , 1
     }
-  }
-  args '-rf', format
-  args '-rff', resultFile
+    args '-foe', 'true'   //fail-on-error
+    args '-v', 'NORMAL'   //verbosity [SILENT, NORMAL, EXTRA]
+    if(profilers) {
+      profilers.split(',').each {
+        args '-prof', it
+      }
+    }
+    args '-jvmArgsPrepend', '-Xmx3072m'
+    args '-jvmArgsPrepend', '-Xms3072m'
+    if(jvmArgs) {
+      for(jvmArg in jvmArgs.split(' ')) {
+        args '-jvmArgsPrepend', jvmArg
+      }
+    }
+    args '-rf', format
+    args '-rff', resultFile
 
-  doFirst {
     println "\nExecuting JMH with: $args \n"
     resultFile.parentFile.mkdirs()
+    super.exec()
   }
-}
\ No newline at end of file
+}
+
+task jmh(type: JmhExec, description: 'Executing JMH benchmarks') {
+  classpath = sourceSets.main.runtimeClasspath + configurations.current
+}
+
+task jmhBaseline(type: JmhExec, description: 'Executing JMH baseline benchmarks') {
+  classpath = sourceSets.main.runtimeClasspath + configurations.baseline
```

Comment:
see above: set the result file to a distinctly "baseline version" one.
the task should discard an existing result file ONLY if it matches the "baseline version" pattern (ie, don't let this task replace the file of the main one)

Reply:
I guess this one is also resolved?